go/mysql/server_test.go

```diff
@@ -917,22 +912,19 @@ func TestTLSRequired(t *testing.T) {
 	// Below, we are enabling --ssl-verify-server-cert, which adds
 	// a check that the common name of the certificate matches the
 	// server host name we connect to.
-	l, err := NewListener("tcp", ":0", authServer, th, 0, 0, false)
+	l, err := NewListener("tcp", "127.0.0.1:", authServer, th, 0, 0, false)
```

Comment:
I've updated quite few tests to this style. Binding on the local ip means we're less environment dependent and it removes the need further down to use the hostname. We use the option to pass in a separate `ServerName` so we still have TLS validation working, but don't depend as much on environment things that could be more brittle.